### PR TITLE
Iv write

### PR DESCRIPTION
**Pull request by jemmaissroff**

### CODE REVIEW

File: yjit/src/codegen.rs [hunk 1]
```diff
@@ -2134,33 +2137,246 @@ fn gen_getinstancevariable(
...
+                gen_write_iv(asm, comptime_receiver, recv, ivar_index, write_val, needs_extension);
+
+                asm.comment("write shape");
+
+                let shape_bit_size = unsafe { rb_shape_id_num_bits() }; // either 16 or 32 depending on RUBY_DEBUG
+                let shape_byte_size = shape_bit_size / 8;
+                let shape_opnd = Opnd::mem(shape_bit_size, recv, RUBY_OFFSET_RBASIC_FLAGS + (8 - shape_byte_size as i32));
+
+                // Store the new shape
+                asm.store(shape_opnd, Opnd::UImm(0 as u64));
```
Comment (k0kubun) [inline yjit/src/codegen.rs:2342]: This is no longer needed. I actually pushed a commit to remove this but it seems to be lost during the rebase.

File: yjit/src/cruby_bindings.inc.rs [hunk 2]
```diff
@@ -234,6 +234,9 @@ pub const RUBY_FL_SINGLETON: ruby_fl_type = 4096;
 pub type ruby_fl_type = i32;
 pub type st_data_t = ::std::os::raw::c_ulong;
 pub type st_index_t = st_data_t;
+extern "C" {
```
Comment (jemmaissroff) [inline yjit/src/cruby_bindings.inc.rs:237]: I accidentally dropped this in my latest rebase off master, so putting it back

File: shape.c [hunk 3]
```diff
@@ -306,6 +306,24 @@ rb_shape_id_num_bits(void)
...
+uint8_t
+rb_shape_flag_shift(void)
+{
+    return SHAPE_FLAG_SHIFT;
+}
+
+uint64_t
+rb_shape_flag_mask(void)
+{
+    return SHAPE_FLAG_MASK;
```
Comment (k0kubun) [inline shape.c:324]: Can't we import `SHAPE_FLAG_SHIFT` and `SHAPE_FLAG_MASK` directly?

**Comment by jemmaissroff** [comment]
This draft was a proof of concept for #6767 , which we've now merged

**Pull request closed without merging**